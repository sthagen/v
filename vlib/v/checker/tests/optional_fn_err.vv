fn foo() ? {
	println('foo is called')
}

fn bar() ?int {
	return none
}

fn main() {
	// Calling foo() without ? or an or block, should be an error.
	foo()

	_ := bar()
	_ := [bar()]
}
