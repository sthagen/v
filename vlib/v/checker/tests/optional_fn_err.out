vlib/v/checker/tests/optional_fn_err.vv:11:2: error: foo() returns an option, so it should have either an `or {}` block, or `?` at the end
    9 | fn main() {
   10 |     // Calling foo() without ? or an or block, should be an error.
   11 |     foo()
      |     ~~~~~
   12 | 
   13 |     _ := bar()
vlib/v/checker/tests/optional_fn_err.vv:13:7: error: bar() returns an option, so it should have either an `or {}` block, or `?` at the end
   11 |     foo()
   12 | 
   13 |     _ := bar()
      |          ~~~~~
   14 |     _ := [bar()]
   15 | }
vlib/v/checker/tests/optional_fn_err.vv:14:8: error: bar() returns an option, so it should have either an `or {}` block, or `?` at the end
   12 | 
   13 |     _ := bar()
   14 |     _ := [bar()]
      |           ~~~~~
   15 | }
